overlord/snapstate/handlers.go

```diff
@@ -834,16 +834,14 @@ func (m *SnapManager) doUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	if experimentalRefreshAppAwareness {
-		// A process may be created after the soft refresh done upon
-		// the request to refresh a snap. If such process is alive by
-		// the time this code is reached the refresh process is stopped.
-		// In case of failure the snap state is modified to indicate
-		// when the refresh was first inhibited. If the first
-		// inhibition is outside of a grace period then refresh
-		// proceeds regardless of the existing processes.
-		if err := inhibitRefresh(st, snapst, oldInfo, HardNothingRunningRefreshCheck); err != nil {
+		// Perform a hard refresh check. Upon success the returned lock can be
```

Comment:
the check now also does an inhibit so calling it just a check is a bit odd

Reply:
I've renamed it to hard refresh _flow_ now.